### PR TITLE
Added datatable proof of concept

### LINKED ISSUE #613: DataTable.js
As a user, I'd like a way to view and sort by terms and other fields on assets, without having to open the learning resource panel. 

We've had many requests to use DataTable.js, possibly as an alternate "curator" view of the learning resource listing page.

### PR DESCRIPTION
**Pull request by noisecapella**
Fixes #613

### CODE REVIEW

**Comment by noisecapella** [comment]
This is ready for review once #659 is merged

**Comment by pwilkins** [comment]
Increasing the page size using the `Page Size` widget to 200 at the bottom of the datatable works, but choosing a smaller size once a larger size has been selected leaves the datatable empty.

**Comment by pwilkins** [comment]
Consider adding the breadcrumb text as a datatable column, for example, `Advanced Introductory Classical Mechanics / 1: Newton's Laws of Motion / Newton's First Law / Inertial and Non Inertial Frame of References`.  It would be a useful sort field.

**Comment by noisecapella** [comment]
@pwilkins That should be available under Description Path. You will probably have to resize the column, you can do that by clicking on the right edge in the header and dragging it.

**Comment by noisecapella** [comment]
I can't reproduce the `Page Size` problem you mentioned. Can you note what the current page size was at each point?

**Comment by pwilkins** [comment]
Steps to reproduce `Page Size` problem: 
Starting with an initial page size of 20, 
Change the page size to 200,
Scroll to the bottom and change the page size back to 20.

This occurs in FireFox.

**Comment by noisecapella** [comment]
I see the problem now, I'll look into it

File: ui/static/ui/js/lore_data_table.jsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+          name: "title",
+          title: "Title",
+        },
+        {
+          name: "description",
+          title: "Description"
+        },
+        {
+          name: "description_path",
+          title: "Description Path",
```
Comment (pdpinch) [inline ui/static/ui/js/lore_data_table.jsx]: Can we label this "Path" or "Learning Resource Path" if you don't think that's clear enough?

"Description Path" doesn't make sense to me.

File: ui/static/ui/js/lore_data_table.jsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+          name: "course",
+          title: "Course"
+        },
+        {
+          name: "run",
+          title: "Run"
+        },
+        {
+          name: "xa_nr_views",
+          title: "Number of Views",
```
Comment (pdpinch) [inline ui/static/ui/js/lore_data_table.jsx]: Shorten to "Views"

File: ui/static/ui/js/lore_data_table.jsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+          type: "number"
+        },
+        {
+          name: "xa_nr_attempts",
+          title: "Number of Attempts",
```
Comment (pdpinch) [inline ui/static/ui/js/lore_data_table.jsx]: shorten to "Attempts"

File: ui/static/ui/js/require_config.js [hunk 4]
```diff
@@ -5,8 +5,9 @@ var REQUIRE_PATHS = {
   bootstrap: "bootstrap/dist/js/bootstrap.min",
   icheck: "icheck/icheck.min",
   retina: "retina.js/dist/retina.min",
-  react: "react/react-with-addons.min",
+  React: "react/react-with-addons",
```
Comment (pdpinch) [inline ui/static/ui/js/require_config.js]: Why did you revert the minification?

Comment (noisecapella) [inline ui/static/ui/js/require_config.js]: That was unintentional, I'll change it back

**Comment by pdpinch** [comment]
Just my 3 comments. Otherwise this looks good to me as a proof of concept. 

@Ferdi ?

**Comment by pwilkins** [comment]
@noisecapella I didn't immediately recognize that `Description Path` was the same as what I was calling "breadcrumbs".  Now that I look at it, it's not as useful in practice as I had imagined.

**Comment by noisecapella** [comment]
@pwilkins I filed an issue for the scrolling problem, given that it's a proof of concept I won't hold up the merge for that though

**Comment by pwilkins** [comment]
@noisecapella I agree.  This shouldn't delay the merge.

**Comment by Ferdi** [comment]
I had the impression we would specifically use datatable.js library.   Do we know how these 2 libraries (react-datagrid vs datatablejs) compare in terms of features?  My only concern is that the users who requested these feature might expect some specific features from datatablejs.

**Comment by jamiefolsom** [comment]
Hi @Ferdi, 

My understanding (@noisecapella, correct me if I'm wrong) is that datatable.js is one of several libraries that offer similar functionality, and is more mature than the others. 

However, while it's possible to use Datatable,js alongside (and separate from) react components, it's not possible to use it within a react component without losing the features that make it better than the alternatives, since it modifies the DOM directly, which is in conflict with the react approach.

**Pull request merged**